### PR TITLE
feat(providers): Create IsReadOnly API for blob storage

### PR DESCRIPTION
**Pull request by ashmrtn**
Create an API that allows higher-layers to determine if the storage is in read-only mode or not. This can then be leveraged by higher-layers to determine if they should execute operations like manifest compaction

related to:
* #3225
* #3226

### CODE REVIEW

**Comment by jkowalski** [comment]
we keep adding more and more of these methods, instead of repeating all this implementation can we rename`UnsupportedBlobRetention` to something like`DefaultProviderImplementation` which will provide sensible default implementations for less-commonly implemented methods:

* `IsReadOnly` - returns false
* `ExtendBlobRetention` - returns `ErrUnsupportedObjectLock`
* `Close` - noop
* `FlushCaches` - noop

perhaps also `GetCapacity` - returns `ErrNotAVolume`

All providers will embed `DefaultProviderImplementation` and they are free to override just the methods they care about.

**Comment by jkowalski** [comment]
also can this PR be only about `blob.Storage` changes and not include any `content`-level changes?

**Comment by julio-lopez** [comment]
> All providers will embed `DefaultProviderImplementation` and they are free to override just the methods they care about.

Separate PR though.

File: repo/content/committed_read_manager.go [hunk 1]
```diff
@@ -112,6 +112,12 @@ type SharedManager struct {
 	metricsStruct
 }
 
+// IsReadOnly returns whether this instance of the SharedManger supports
+// mutations to the index.
+func (sm *SharedManager) IsReadOnly() bool {
+	return sm.st.IsReadOnly()
+}
+
```
Comment (julio-lopez) [inline repo/content/committed_read_manager.go:120]: Remove content-level change to keep changes contained to the blob layer

```suggestion
```

**Review by julio-lopez — APPROVED** [review]
LGTM modulo suggested change. Feel free to merge after applying the suggestion.

**Comment by jkowalski** [comment]
> > All providers will embed `DefaultProviderImplementation` and they are free to override just the methods they care about.
> 
> Separate PR though.

Can we simply rename `UnsupportedBlobRetention` => `DefaultProviderImplementation` and add only `IsReadOnly` there, the rest of the refactoring can be in separate PR (although I'm fine do to it in one PR, since we'd be only removing stub methods)

**Pull request merged**